message/validation/utils_test.go

```diff
@@ -147,75 +146,68 @@ func TestValidateConsensusMessage(t *testing.T) {
 		{
 			name: "ProposalMessage_ExceedsLimit_ReturnsError",
 			input: input{
-				counts:           &MessageCounts{Proposal: 2},
+				seenMsgTypes:     &SeenMsgTypes{v: 0b10},
 				signedSSVMessage: &spectypes.SignedSSVMessage{},
 				msg:              &specqbft.Message{MsgType: specqbft.ProposalMsgType},
-				limits:           MessageCounts{Proposal: 0},
 			},
-			expectedError: fmt.Errorf("message is duplicated, got proposal, having pre-consensus: 0, proposal: 2, prepare: 0, commit: 0, round change: 0, post-consensus: 0"),
+			expectedError: fmt.Errorf("message is duplicated, got proposal, having pre-consensus: 0, proposal: 1, prepare: 0, commit: 0, round change: 0, post-consensus: 0"),
 		},
 		{
 			name: "PrepareMessage_ExceedsLimit_ReturnsError",
 			input: input{
-				counts:           &MessageCounts{Prepare: 2},
+				seenMsgTypes:     &SeenMsgTypes{v: 0b100},
 				signedSSVMessage: &spectypes.SignedSSVMessage{},
 				msg:              &specqbft.Message{MsgType: specqbft.PrepareMsgType},
-				limits:           MessageCounts{Prepare: 0},
 			},
-			expectedError: fmt.Errorf("message is duplicated, got prepare, having pre-consensus: 0, proposal: 0, prepare: 2, commit: 0, round change: 0, post-consensus: 0"),
+			expectedError: fmt.Errorf("message is duplicated, got prepare, having pre-consensus: 0, proposal: 0, prepare: 1, commit: 0, round change: 0, post-consensus: 0"),
 		},
 		{
 			name: "CommitMessageWithSingleOperator_ExceedsLimit_ReturnsError",
 			input: input{
-				counts:           &MessageCounts{Commit: 2},
+				seenMsgTypes:     &SeenMsgTypes{v: 0b1000},
 				signedSSVMessage: &spectypes.SignedSSVMessage{OperatorIDs: []spectypes.OperatorID{1}},
 				msg:              &specqbft.Message{MsgType: specqbft.CommitMsgType},
-				limits:           MessageCounts{Commit: 0},
 			},
-			expectedError: fmt.Errorf("message is duplicated, got commit, having pre-consensus: 0, proposal: 0, prepare: 0, commit: 2, round change: 0, post-consensus: 0"),
+			expectedError: fmt.Errorf("message is duplicated, got commit, having pre-consensus: 0, proposal: 0, prepare: 0, commit: 1, round change: 0, post-consensus: 0"),
 		},
 		{
 			name: "RoundChangeMessage_ExceedsLimit_ReturnsError",
 			input: input{
-				counts:           &MessageCounts{RoundChange: 2},
+				seenMsgTypes:     &SeenMsgTypes{v: 0b10000},
 				signedSSVMessage: &spectypes.SignedSSVMessage{},
 				msg:              &specqbft.Message{MsgType: specqbft.RoundChangeMsgType},
-				limits:           MessageCounts{RoundChange: 0},
 			},
-			expectedError: fmt.Errorf("message is duplicated, got round change, having pre-consensus: 0, proposal: 0, prepare: 0, commit: 0, round change: 2, post-consensus: 0"),
+			expectedError: fmt.Errorf("message is duplicated, got round change, having pre-consensus: 0, proposal: 0, prepare: 0, commit: 0, round change: 1, post-consensus: 0"),
```

Comment:
This new error message makes it a bit harder to understand what's going on because it sort of claims we've seen 1 round change in that particular test-case (not 2),

can we maybe improve this a bit ? Eg. to print `true/false` if keeping track of exact number of times is memory-heavy